File: ts/setZeroes.ts
/**
 Do not return anything, modify matrix in-place instead.
 */
function setZeroes(matrix: number[][]): void {
  const x: number[] = [];
  const y: number[] = [];

  matrix.forEach((row, i) => {
    row.forEach((col, j) => {
      if (col === 0) {
        x.push(i);
        y.push(j);
      }
    });
  });

  for (let i = 0; i < matrix.length; i++) {
    for (let j = 0; j < matrix[0].length; j++) {
      if (x.includes(i) || y.includes(j)) {
        matrix[i][j] = 0;
      }
    }
  }
}

export default () => {
  const matrix = [
    [1, 1, 1],
    [1, 0, 1],
    [1, 1, 1],
  ];
  setZeroes(matrix);
  console.log(matrix);
};

File: rs/src/set_zeroes.rs
fn set_zeroes(matrix: &mut Vec<Vec<i32>>) {
    let mut x: Vec<usize> = vec![];
    let mut y: Vec<usize> = vec![];

    for (i, row) in matrix.iter().enumerate() {
        for (j, col) in row.iter().enumerate() {
            if col.to_owned() == 0 {
                x.push(i);
                y.push(j);
            }
        }
    }

    for i in 0..matrix.len() {
        for j in 0..matrix[i].len() {
            if x.contains(&i) || y.contains(&j) {
                matrix[i][j] = 0;
            }
        }
    }
}

pub fn main() {
    let mut matrix = [vec![1, 1, 1], vec![1, 0, 1], vec![1, 1, 1]].to_vec();
    set_zeroes(&mut matrix);
    println!("{:?}", matrix);
}
